bokeh/cli/core.py

```diff
@@ -164,24 +242,38 @@ def update_source(self, new_source):
         ns.index = [len_source]
         self.source = pd.concat([self.source, ns])
 
-        _c = create_chart(self.series, ns, self.index, self.factories, **self.chart_args)
-        _c._setup_show()
-        _c._prepare_show()
-        _c._show_teardown()
-
-        for k, v in _c.source.data.items():
-            self.chart.source.data[k] = list(self.chart.source.data[k]) + list(v)
-
-        chart = self.chart.chart
-        plot = chart.plot
-        plot.y_range.end = max(
-            plot.y_range.end, _c.chart.plot.y_range.end
-        )
-        plot.y_range.start = min(
-            plot.y_range.start, _c.chart.plot.y_range.start
-        )
-        chart.session.store_objects(self.chart.source)
-        chart.session.store_objects(plot)
+        # TODO: This should be replaced with something that just computes
+        #       the new data and source
```

Comment:
I think so. One thing to keep in mind is that even if we are creating the whole chart now we are doing only for the new data ... not the entire source. Also CLI now provides window_size option which allow to limit the size of the data...